src/DataSet/ScalarDataSet.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\DataSet;
+
+use Yiisoft\Validator\DataSetInterface;
+
+final class ScalarDataSet implements DataSetInterface
```

Comment:
What is the use case for this one? It looks a bit out of place.

Reply:
[It's used for strings, numbers and other scalar types](https://github.com/yiisoft/validator/pull/116/files/f26c7594b8275bb1002476dd6a5425f4628bba12#diff-9dad0f0dabc730596b3399e6fe25aed903dba8bc7836fe778fce387cf9f3e111R62)